packages/web/contexts/GameContext.tsx

```diff
@@ -0,0 +1,328 @@
+      // const data = (await response.json()) as GameProperties;
+      const data = gameJson as GameProperties;
+      console.log('fetchGameData', data);
+
+      if (data) {
+        setGameDataState(data);
+      }
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error);
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (address) {
+      setAccount(address);
+    }
+  }, [address]);
+
+  const game = useMemo((): GameProperties => {
+    try {
+      if (!isLoading && gameDataState) {
+        const {
+          assets,
+          name,
+          startingElement,
+          elements,
+          connections,
+          components,
+        } = gameDataState;
+        return {
+          name,
+          assets,
+          startingElement,
+          elements,
+          connections,
+          components,
+        } as GameProperties;
+      }
+      return {
+        name: '',
+        assets: {},
+        startingElement: '',
+        elements: {},
+        connections: {},
+        components: {},
+      } as GameProperties;
+    } catch (error) {
+      console.log('fetchGameData error: ', { error });
+      return {
+        name: '',
+        startingElement: '',
+        assets: {},
+        elements: {},
+        connections: {},
+        components: {},
+      } as GameProperties;
+    }
+  }, [gameDataState, isLoading]);
+
+  /** Function to get, set or clear game state
+   *
+   * @param currentPlace - the current place of the player
+   * @param reset - if true, clear the game state
+   * @returns the game state or undefined if reset is true or the game state is not set
+   */
+  const gameState = (
+    currentPlace?: string,
+    reset?: boolean,
+  ): IGameState['state'] => {
+    if (currentPlace !== undefined) {
+      set('OnboardingGameState', currentPlace);
+    }
+    if (reset) {
+      remove('OnboardingGameState');
+      remove('OnboardingGameVisitedElements');
+    }
+
+    const state = get('OnboardingGameState');
+    return state;
+  };
+
+  /** Callback function to handle the users choices.
+   *
+   *  Takes the connected element id from the element's output and routes to the next step
+   */
+  const handleChoice = useCallback(
+    async (target: string): Promise<string | undefined> => {
+      try {
+        if (target) gameState(target);
+        const success = gameState() === target;
+        if (success) {
+          return target;
+        }
+        throw new Error('Game progression failed');
+      } catch (error) {
+        // add toast here
+        console.log(error);
+        return undefined;
+      }
+    },
+    [],
+  );
+
+  /** Increment number of elements a user has visited & store in localStorage
+   * This is used to trigger an achievement after so many elements have been visited
+   */
+  const visitedElements = (increment?: boolean): string => {
+    const visited: string | null = get('OnboardingGameVisitedElements');
```

Comment:
Having this as a string representing a number is a kludge & I'd convert it from a string as soon as you pull it out of storage.

```suggestion
    const visited: number | null = Number(store.get('OnboardingGameVisitedElements'));
```